ptp/ptp4u/server/server.go

```diff
@@ -111,9 +111,9 @@ func (s *Server) Start() error {
 	// Drain check
 	go func() {
 		defer wg.Done()
-		for {
-			<-time.After(s.Config.DrainInterval)
-
+		ticker := time.NewTicker(s.Config.DrainInterval)
+		defer ticker.Stop()
+		for ; true; <-ticker.C {
```

Comment:
I don’t think you need a «true» statement here

Reply:
otherwise the first tick won't happen until after the draininterval, just like now with time.After.